packages/qwik-city/src/middleware/request-handler/resolve-request-handlers.ts

```diff
@@ -528,6 +536,39 @@ export async function handleRedirect(requestEv: RequestEvent) {
   }
 }
 
+export async function handleRewrite(requestEv: RequestEvent) {
+  const isPageDataReq = requestEv.sharedMap.has(IsQData);
+  if (!isPageDataReq) {
+    return;
+  }
+  try {
+    await requestEv.next();
+  } catch (err) {
+    if (!(err instanceof RewriteMessage)) {
+      throw err;
+    }
+  }
+
+  if (requestEv.headersSent) {
+    return;
+  }
+
+  const status = requestEv.status();
+  const location = requestEv.headers.get('Rewrite-Location');
+  const isRewrite = status === 200 && location;
+
+  if (isRewrite) {
+    const adaptedLocation = makeQDataPath(location);
+
+    if (adaptedLocation) {
+      requestEv.headers.set('Rewrite-Location', adaptedLocation);
```

Comment:
Not sure if this is necessary?

Reply:
I took it from the redirect flow as well, and it is necessary actually, but I may be able to do it differently.
It signals the request flow that it should fetch the q-data.json and not the page itself.

I'll see if I can remove the header and make it happen in an alternative way.